app/deploy.go

```diff
@@ -195,7 +195,7 @@ func (o *DeployOptions) GetKind() (kind DeployKind) {
 		}
 		return DeployUpload
 	}
-	if o.Commit != "" {
+	if o.Commit != "" && o.ArchiveURL == "" {
```

Comment:
Commit deploy can be made by ArchiveURL and ArchiveFile methods.
The difference is the method git pre-receive hook uses.

Reply:
@gfleury you can check for o.Commit and for app Token. If have both so it's of Kind DeployGit.

Reply:
Hi @guilhermebr, correct if I'm wrong, but this was the "behavior" before since o.Commit was being set to empty if the token wasn't from an app. And then rely only if commit id was empty or not to define if it was from kind DeployGit. 
Checking the token inside getKind() seems a different from how it was being done before, as getKind only rely on the DeploymentOptions to define the kind. (DeploymentOptions were being defined to force a Kind). I'm not sure how is the best way. The way I've done is following the same 'idea' from before.